tests/integration/test_requirements_validation.py

```
@@ -0,0 +1,358 @@
+
+Tests ensure:
+1. Proper dependency specification format
+2. Version pinning and security
+3. Compatibility with existing dependencies
+4. No conflicting versions
+"""
+
+import pytest
+import re
+from pathlib import Path
+from typing import List, Tuple, Dict
```

Ruff (0.14.7)

`typing.List` is deprecated, use `list` instead

(UP035)